zulipterminal/ui_tools/boxes.py

```diff
@@ -109,14 +109,17 @@ def generic_autocomplete(self, text: str, state: int) -> Optional[str]:
                 typeaheads = autocomplete_func(text, prefix)
                 reduced_typeaheads = typeaheads[:num_suggestions]
                 try:
-                    return reduced_typeaheads[state]
+                    typeahead = reduced_typeaheads[state]
                 except (IndexError, TypeError):
-                    return None
+                    typeahead = [None, None]  # type: ignore
```

Comment:
Is the type ignore necessary due to the nested tuples?

Can we just return `None` here as before and handle the success below the try block?

Reply:
We can't return `None` here because we still need to set the footer when the exception is raised. The exception is raised when a user has cycled through all the suggestions and returns back to the non-completed state.(Or goes back using `ctrl+r`)

Reply:
The tuples are of the form `('#**announce**', 'announce')`. I need to set the type to `Tuple[Optional[str], Optional[str]]` which doesn't fit the line so I added type ignore. mypy doesn't allow line breaks for type declarations(https://github.com/python/mypy/issues/4511)

But I changed it from list to tuple for consistency.

Reply:
Ah, I see. So you want to call `set_footer_text( )` from this method itself instead of going to `set_typeahead_footer( )`? Well, that's certainly not a problem - only thing would be the `  [more]  ` addition would be duplicated.